src/pipelines/token_classification.rs

```diff
@@ -1129,12 +1130,18 @@ impl TokenClassificationModel {
             }
             LabelAggregationOption::Mode => {
                 let counts = tokens.iter().fold(HashMap::new(), |mut m, c| {
-                    *m.entry((c.label_index, c.label.as_str())).or_insert(0) += 1;
+                    let entry = m
+                        .entry((c.label_index, c.label.as_str()))
+                        .or_insert((0, 0.));
+                    *entry = (
+                        entry.0 + 1,
+                        if c.score > entry.1 { c.score } else { entry.1 },
+                    );
                     m
                 });
                 counts
                     .into_iter()
-                    .max_by(|a, b| a.1.cmp(&b.1))
+                    .max_by_key(|item| (item.1 .0, OrderedFloat(item.1 .1)))
```

Comment:
I found this `.1 .0` syntax pretty hard to read - I wonder if destructuring here would be easier to read (and symmetric with the `map` on the next line):
```suggestion
                    .max_by_key(|&(_, (count, score))| (count, OrderedFloat(score)))
```